### ISSUE TITLE
Clipboard isn't converted when surrounded by quotes

### ISSUE THREAD

**Issue by Banaanae** [mmikeww/AHK-v2-script-converter#163]
V1:
```ahk
Clipboard := """" Trim(Clipboard) """"
```

V2:
```ahk
A_Clipboard := "`"" Trim(Clipboard) "`""
```

V2E:
```ahk
A_Clipboard := "`"" Trim(A_Clipboard) "`""
```

Could either by caused by @andymbody string masking
Or my #111 fix

<sub>I know this is a very niche issue but happened while converting [this repo](https://github.com/Haris00911/QuickQuotes)</sub>

**Comment by andymbody** [comment]
Yes... this is due to the masking... gonna have to play with the needle to see if this can be avoided, while still preventing others from being converted.

**Comment by andymbody** [comment]
The current needle will cause this. I changed it from my original. I tested this with my original needle and I does not seem to cause this when I run it alone, but when run within the conversion script, it still occurs. So there may be a could a couple contributing factors. In other words, I'm not sure adjusting the needle alone will solve it, but I will see.

**Comment by andymbody** [comment]
Yes the current needle of the mask will cause this... change it to (line 3734)
pattern := '(?<!")"[^"]+"(?!")' 
Or should I do another pull-request for this?

**Comment by andymbody** [comment]
Yes the current mask needle will cause this... I changed it from my original.

Change it back to this (line 3734)
pattern := '(?<!")"[^"]+"(?!")'

then.. try this example
Clipboard := """" Trim(Clipboard) """"
Clipboard := "clipboard"

Or should I submit another pull-request for this?

**Comment by andymbody** [comment]
Yes the current mask needle will cause this... I changed it from my original.

Change it to this (line 3734)
pattern := '(?<!")"[^"]+"(?!")'

then.. try this example
Clipboard := """" Trim(Clipboard) """"
Clipboard := "clipboard"

Or should I submit another pull-request for this?

**Comment by Banaanae** [comment]
That fixes it
I could commit it but I'll let you get the practice with git
When you make your PR also move `Assignment_ex8.ah1` and `Assignment_ex8.ah2` from failing tests to the String folder
You can also include `fix #163` somewhere in the commit or PR to automatically close this issue when merged

**Comment by andymbody** [comment]
> That fixes it I could commit it but I'll let you get the practice with git When you make your PR also move `Assignment_ex8.ah1` and `Assignment_ex8.ah2` from failing tests to the String folder You can also include `fix #163` somewhere in the commit or PR to automatically close this issue when merged

lol... ok... i do need the practice I suppose. Thanks for the instructions and holding my hand while I learn.

**Comment by andymbody** [comment]
This fix is a work around "for now", but I will be adding a better fix soon. This better fix will mask double quotes prior to masking full strings. That will allow variables to remain intact. And should also prevent other unique quote configurations from causing an issue. If you want to wait for that fix, you can reject this one and #164 commit (which fixes some typos that I created by mistake). The new fix will be overwriting the changes made in #163, and #164.

**Comment by andymbody** [comment]
Will submit changes when I have more experience with Git.
